### DISCUSSION TITLE
Sea-ORM still includes sqlx-mysql dependency even when no mysql feature is enabled

### DISCUSSION THREAD

**Discussion by DenuxPlays** [SeaQL/sea-orm#2172] [Q&A]
## Description

I've added sea-orm like this:
```toml
sea-orm = { version = "1.0.0-rc.2", features = ["sqlx-postgres", "runtime-tokio-rustls", "macros", "mock"] }
```
And the sqlx-mysql dependency can still be found in the Cargo.lock file:
![image](https://github.com/SeaQL/sea-orm/assets/83671398/60fd93a2-46ad-4033-9192-538688e0ebab)

## Steps to Reproduce

1. add sea-orm like mentioned above
2. search for sqlx-mysql in your Cargo.lock file

### Expected Behavior

Does not include this as a dependency

### Actual Behavior

Includes it...

### Reproduces How Often

Always

### Workarounds

None known

## Reproducible Example

https://github.com/financrr/financrr-app/tree/main/backend

## Versions

v1.0.0-rc.2

**Comment by tyt2y3** [comment]
I am not seeing this behaviour on our repo's examples. May be some crate in the workspace of your repo enabled it?

**Reply by DenuxPlays** [reply]
I don't think so.
I only communicate to postgres via SeaOrm or SeaMigration.
Also regenerated the lock file.
cargo tree supports this:
![image](https://github.com/SeaQL/sea-orm/assets/83671398/a89b9100-fc92-4a96-8883-166ecf7ded7e)

**Reply by DenuxPlays** [reply]
Maybe this has something to do with it?
![image](https://github.com/SeaQL/sea-orm/assets/83671398/47b3dd8f-99b2-45d3-97b9-56e1e7247c88)

Shouldn't be `sqlx/postgres` enough?

**Reply by yguenduez** [reply]
I also had the same issue with mysql, as `cargo audit` even reported a security issue on this mysql dependency.
I got rid of it by just disabling the `default-features`.
(But I also do not use the release candidate now - and use sqlite). 

Maybe this helps you (disabling default-features):

```
sea-orm = { version = "0.12.15", features = ["sqlx-sqlite", "runtime-tokio-rustls", "macros" ], default-features = false }
```

**Comment by SZenglein** [comment]
Pretty sure this is a cargo bug. Multiple people have run in to this issue with sqlx in particular. `cargo tree` will correctly not list sqlx-mysql, but it's still pulled into Cargo.lock, which is read by cargo-audit.

https://github.com/rust-lang/cargo/issues/10801

**Comment by sam0x17** [comment]
this bug comes back if you add any of the "with-xxxxx" features it seems
